src/site/markdown/in_depth_walkthrough.md

```diff
@@ -0,0 +1,316 @@
+to a plugin for the synthesis of the code responsible for the parameter extraction.  This will become clear soon,
+but before continuing, a brief note.
+
+#### A Note on the Power of Imports
+
+It is important to note that the ability to import is **extremely powerful**!  By allowing imports, Aloha can 
+easily be extended outside the library.  Imports provide the power analogous to 
+[user-defined functions](https://en.wikipedia.org/wiki/User-defined_function) (UDFs) in databases, but imports 
+are actually much more powerful due to scala's facility for 
+[implicit parameters](http://daily-scala.blogspot.com/2010/04/implicit-parameters.html).
+
+
+>
+> Imports can do more than just add language features. They can *change the meaning of the language*.
+>
+
+
+Imports are specified globally in CompiledSemantics, away from where the features are specified. This is by design 
+so that the same specifications can have different meaning in different problem domains.  The consequence is that 
+imports need to be used with caution, especially when referencing implicit language features.  The following example 
+illustrates this.  
+
+Imagine a third party developer wrote the following code 
+
+```scala
+package com.fictitious.company
+
+object Ops {
+  /**
+   * @param v an explicit value
+   * @param a an implicit value to add to v 
+   * @param ring A type class representing an algebraic ring.
+   * @tparam A the type of algebraic ring
+   * @return ''v'' added to some value ''a'' in the implicit scope
+   */
+  def add[A](v: A)(implicit a: A, ring: Numeric[A]) = num.plus(v, a)
+}
+
+object Implicits {
+  implicit val two: Int = 2 
+  implicit val four: Int = 4
+}
+```
+
+Let’s assume a feature specification: `add(${some.value})` and the parameter 
+`${some.value}` has an integer value, *1*. If the imports were: 
+
+- `com.fictitious.company.Ops.add`
+- `com.fictitious.company.Implicits.two`
+
+Then the output of the feature would be *3* in this instance.  If we changed the imports to 
+
+- `com.fictitious.company.Ops.add`
+- `com.fictitious.company.Implicits.four`
+
+then the **same feature specification** would yield a value of *5*.  While this can be extremely useful, it can also
+be extremely dangerous.  It is imperative to understand the imports being imported into scope.
+
+
+### Protobuf Plugin 
+
+Currently there are two main CompiledSemantics plugins.  The first is 
+[CompiledSemanticsProtoPlugin](aloha-core/scaladocs/index.html#com.eharmony.aloha.semantics.compiled.plugin.proto.CompiledSemanticsProtoPlugin) located 
+in the `com.eharmony.aloha.semantics.compiled.plugin.proto` package in [aloha-core](aloha-core/index.html). This 
+plugin allows for arbitrary kinds of [Protocol Buffer](https://developers.google.com/protocol-buffers/) instances 
+that extend [GeneratedMessage](https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/GeneratedMessage).
+
```

Comment:
Can we add another sentence here about how protobufs are typesafe so they make integrating with Aloha really easy.  Also add something here about the protobuf versioning problems.  Finally add a plug for Thrift and Avro.  Something like "these are two other common communication formats that would be great for Aloha to support in the future, and it would be great if someone wants to contribute these to the project."